Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MultiPositionController.cs

```diff
@@ -0,0 +1,1001 @@
+                        }
+                        if (Locomotive.DynamicBrakePercent > 1)
+                        {
+                            Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent - 1);
+                        }
+                        if (CanControlTrainBrake)
+                        {
+                            if (Locomotive.TrainBrakeController.GetStatus().ToLower() != "release")
+                            {
+                                Locomotive.StartTrainBrakeDecrease(null);
+                            }
+                            else
+                                Locomotive.StopTrainBrakeDecrease();
+                        }
+                        applyPower = false;
+                    }
+                    if (applyPower) Locomotive.CruiseControl.SpeedSelMode = CruiseControl.SpeedSelectorMode.On;
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeIncrease)
+                {
+                    isBraking = true;
+                    previousDriveModeWasAddPower = false;
+                    Locomotive.CruiseControl.SpeedSelMode = CruiseControl.SpeedSelectorMode.Neutral;
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "apply")
+                        {
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "neutral")
+                        {
+                            Locomotive.StopTrainBrakeDecrease();
+                        }
+                    }
+                    if (Locomotive.ThrottlePercent < 1 && Locomotive.DynamicBrakePercent < 100)
+                    {
+                        if (Locomotive.DynamicBrakePercent < 0)
+                            Locomotive.DynamicBrakeChangeActiveState(true);
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent + 1f);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeIncreaseFast)
+                {
+                    isBraking = true;
+                    previousDriveModeWasAddPower = false;
+                    Locomotive.CruiseControl.SpeedSelMode = CruiseControl.SpeedSelectorMode.Neutral;
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "apply")
+                        {
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "neutral")
+                        {
+                            Locomotive.StopTrainBrakeDecrease();
+                        }
```

Comment:
5th time, this for sure can be refactored

Reply:
changed to enum comparisons